lib/settings/view/settings_page.dart

```diff
@@ -557,6 +575,102 @@ class _ExposeOnlineSection extends StatelessWidget with WatchItMixin {
+                        Padding(
+                          padding: const EdgeInsets.symmetric(vertical: 4),
+                          child: TextFormField(
+                            controller: lastFmApiKeyController,
+                            decoration: InputDecoration(
+                              hintText: l10n.lastfmApiKey,
+                            ),
+                            validator: (value) {
+                              if (value == null || value.isEmpty) {
+                                return l10n.lastfmApiKeyEmpty;
+                              }
+                              return null;
+                            },
+                            onFieldSubmitted: (value) async {
+                              if (formkey.currentState!.validate()) {
+                                di<SettingsModel>().setLastFmApiKey(value);
+                              }
+                            },
+                          ),
+                        ),
+                        Padding(
+                          padding: const EdgeInsets.symmetric(vertical: 4),
+                          child: TextFormField(
+                            controller: lastFmSecretController,
+                            decoration: InputDecoration(
+                              hintText: l10n.lastfmSecret,
+                            ),
+                            validator: (value) {
+                              if (value == null || value.isEmpty) {
+                                return l10n.lastfmSecretEmpty;
+                              }
+                              return null;
+                            },
+                            onFieldSubmitted: (value) async {
+                              if (formkey.currentState!.validate()) {
+                                di<SettingsModel>().setLastFmSecret(value);
+                              }
+                            },
+                          ),
+                        ),
+                      ],
+                    ),
+                  ),
+              ],
+            ),
+            trailing: Column(
+              mainAxisAlignment: MainAxisAlignment.spaceBetween,
+              children: [
+                CommonSwitch(
+                  value: lastFmEnabled,
+                  onChanged: (v) {
+                    di<SettingsModel>().setEnableLastFmScrobbling(v);
+                  },
+                ),
+                if (lastFmEnabled)
+                  ImportantButton(
+                    onPressed: () async {
+                      if (lastFmApiKeyController.text.isNotEmpty &&
+                          lastFmSecretController.text.isNotEmpty) {
+                        final lastfmua = di<LastFM>() as LastFMUnauthorized;
+                        launchUrl(
+                          Uri.parse(await lastfmua.authorizeDesktop()),
+                        );
+                        await Future.delayed(const Duration(seconds: 20));
```

Comment:
I don't think this is reliant to faliure. what happens if the user somehow aborts the login process?

Reply:
Yeah a webview is not a good idea indeed!

Could you maybe add a retry loop or something?

Reply:
Oh wait, I think last.fm has callback URLs! I'll try to implement using that. This might take a day.

Reply:
The [commit](https://github.com/ubuntu-flutter-community/musicpod/pull/1000/commits/f51d3daefd981303bb25a6085a8ce1cf62695808) addresses this. Please review.

Reply:
I have addressed this by repeatedly polling last.fm every 10 seconds for upto 2 minutes after the button is clicked. 
What's left is to show a dialog that auth failed in case that does not work. I've added a TODO there.